File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int> res(nums.size());
        int l = 0;
        int r = nums.size() - 1;
        int j = nums.size() - 1;
        for(int i = 0; i < nums.size(); i++)
        {
            if(nums[i] < pivot)
            {
                res[l] = nums[i];
                l++;
            }
            if(nums[j] > pivot)
            {
                res[r] = nums[j];
                r--;
            }
            j--;
        }
        while(l <= r)
        {
            res[l] = pivot;
            l++;
        }

        return res;
    }
};

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        res = [0 for _ in range(len(nums))]
        l = 0
        r = len(nums) - 1
        for i in range(len(nums)):
            if nums[i] < pivot:
                res[l] = nums[i]
                l += 1
            if nums[-i - 1] > pivot:
                res[r] = nums[-i - 1]
                r -= 1
        while l <= r:
            res[l] = pivot
            l += 1
        return res